src/wallet/mod.rs

```diff
@@ -307,8 +303,16 @@ where
             FeePolicy::FeeRate(rate) => (*rate, 0.0),
         };
 
-        if builder.send_all && builder.recipients.len() != 1 {
-            return Err(Error::SendAllMultipleOutputs);
+        // try not to move from `builder` because we still need to use it later.
+        let recipients = builder.single_recipient.as_ref().map_or(
+            builder.recipients.iter().map(|(r, v)| (r, *v)).collect(),
+            |recipient| vec![(recipient, 0)],
+        );
+        if builder.single_recipient.is_some() && builder.utxos.is_empty() && !builder.drain_wallet {
```

Comment:
With this logic you are saying that `set_single_recipient` implicitly implies `manually_selected_only` if `builder.utxos` is not empty. I think it would be a good idea to avoid this implicit logic by forcing the caller to either select `manually_selected_only` or `drain_wallet` when doing `set_single_recipient`. 

```suggestion
        if builder.single_recipient.is_some() && !builder.manually_selected_only && !builder.drain_wallet {
```
I would hope that in general when doing `manually_selected_only` with `utxos` empty we have an error of some kind.

Note if you take this suggestion then you should update docs for `set_single_recipient` to make this clear :)

Reply:
Yeah that's a good point!